Sources/TuistCore/Graph/GraphDependencyReference.swift

```diff
@@ -106,11 +106,27 @@ public enum GraphDependencyReference: Equatable, Comparable, Hashable {
         case let (.library(lhsPath, _, _, _), .library(rhsPath, _, _, _)):
             return lhsPath < rhsPath
         case let (.product(lhsTarget, lhsProductName, lhsPlatformFilter), .product(rhsTarget, rhsProductName, rhsPlatformFilter)):
-            return lhsTarget < rhsTarget ||
-                lhsProductName < rhsProductName ||
-                (lhsPlatformFilter != nil && rhsPlatformFilter != nil && (lhsPlatformFilter! < rhsPlatformFilter!))
+            if lhsTarget < rhsTarget {
+                return true
+            } else if lhsTarget == rhsTarget {
+                if lhsProductName < rhsProductName {
+                    return true
+                } else if lhsProductName == rhsProductName {
+                    switch (lhsPlatformFilter, rhsPlatformFilter) {
+                    case let (lhsPlatformFilter?, rhsPlatformFilter?):
+                        return lhsPlatformFilter < rhsPlatformFilter
+                    case (_?, nil):
+                        return true
+                    default:
+                        return false
+                    }
+                }
+            }
+            return false
```

Comment:
niptick: to reduce nesting and improve readibility
```suggestion
            guard lhsTarget != rhsTarget else {
                return lhsTarget < rhsTarget
            }
            
            guard lhsProductName != rhsProductName else {
                return lhsProductName < rhsProductName
            }
            
            switch (lhsPlatformFilter, rhsPlatformFilter) {
            case let (lhsPlatformFilter?, rhsPlatformFilter?):
                return lhsPlatformFilter < rhsPlatformFilter
            case (_?, nil):
                return true
            default:
                return false
            }
```

Reply:
Should be addressed via implementations @adellibovi's suggestion